pennylane_qiskit/noise_models.py

```diff
@@ -0,0 +1,421 @@
+        kraus_ops = [
+            op[0] if isinstance(op, list) else qml.I(0).matrix() for op in error_dict["data"]
+        ]
+        error_dict["data"] = [
+            np.sqrt(prob) * kraus_op for prob, kraus_op in zip(error_probs, kraus_ops)
+        ]
+
+    else:  # pragma: no cover
+        raise ValueError(f"Error {error} could not be converted.")
+
+    error_dict = _process_qerror_dict(error_dict=error_dict)
+
+    return getattr(qml.ops, error_dict["name"])(*error_dict["data"], wires=AnyWires)
+
+
+def _build_noise_model_map(noise_model, **kwargs) -> Tuple[dict, dict]:
+    """Builds noise model maps from which noise model can be constructed efficiently.
+
+    Args:
+        noise_model (qiskit_aer.noise.NoiseModel): Qiskit's noise model
+        kwargs: Optional keyword arguments for providing extra information
+
+    Keyword Arguments:
+        gate_times (Dict[str, float]): gate times for building thermal relaxation error.
+            If not provided, the default value of ``1.0`` will be used for construction.
+        decimals (int): number of decimal places to round the Kraus matrices for errors to.
+            If not provided, the default value of ``10`` is used.
+        atol (float): the relative tolerance parameter. Default value is ``1e-05``.
+        rtol (float): the absolute tolernace parameters. Defualt value is ``1e-08``.
+        optimize (bool): controls if intermediate optimization is used while transforming Kraus
+            operators to a Choi matrix, wherever required. Default is ``False``.
+        multi_pauli (bool): assume depolarization channel to be multi-qubit. This is currently not
+            supported with ``qml.DepolarizationChannel``, which is a single qubit channel.
+
+    Returns:
+        (dict, dict): returns mappings for ecountered quantum errors and readout errors.
+
+    For plugin developers: noise model map tuple consists of following two mappings:
+        * qerror_dmap: noise_operation -> wires -> target_gate
+        * rerror_dmap: noise_operation -> wires -> target_measurement
+    """
+    qerror_dmap = defaultdict(lambda: defaultdict(list))
+
+    # Add default quantum errors
+    for gate_name, error in noise_model._default_quantum_errors.items():
+        noise_op = _build_qerror_op(error, gate_name=gate_name, **kwargs)
+        qerror_dmap[noise_op][AnyWires].append(qiskit_op_map[gate_name])
+
+    # Add specific qubit errors
+    for gate_name, qubit_dict in noise_model._local_quantum_errors.items():
+        for qubits, error in qubit_dict.items():
+            noise_op = _build_qerror_op(error, gate_name=gate_name, **kwargs)
+            qerror_dmap[noise_op][qubits].append(qiskit_op_map[gate_name])
+
+    # TODO: Add support for the readout error
+    rerror_dmap = defaultdict(lambda: defaultdict(list))
+    if noise_model._default_readout_error or noise_model._local_readout_errors:
+        warn("Readout errors are not supported currently and will be skipped.")
+
```

Comment:
@trbromley we have in the requirements that if readout noise is present, PennyLane raises an error. But, I'm wondering if raising a warning instead and _skipping_ the readout noise makes more sense? 🤔